feat: Update the order of system user add/edit form fields and labels and remove NID field

Siyasanga opened the pull request:
Address the issues listed in ticket #6830 like:
- Reordering the field to start with the user's surname
- Rename "First name(s)" label to read "User's first name"
- Rename "Last name" label to read as "User's surname"
- Remove the `NID` field from the user's form
- Write migration script to remove the `NID` value for all existing system Users

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Ask surname first in the User details form

we often receive requests from countries to have the surname first in this form, we have decided to satisfy that requirement by putting surname first for ALL countries.

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Change name and surname labels for User Details form

A request came for the clients to change the labels for "First name(s)" to "User's first name". Also change "Last name" to "User's surname" for clarity and less confussion.

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga commented:
Hi @rikukissa I've removed all references to the `NID` field, can you please have a look when you have some time. I'm currently writing the migration to remove the `NID` field for the already existing system users.

Siyasanga committed:
Remove NID field from the user details form

Remove the question that asks the user for a national ID number in this form as it is not required

https://github.com/opencrvs/opencrvs-core/issues/6830

rikukissa commented on packages/client/src/forms/user/fieldDefinitions/createUser.ts:
```diff
@@ -79,38 +78,38 @@ export function userSectionFormType(): ISerializedFormSection {
         fields: [
           {
-            name: 'firstNamesEng',
+            name: 'familyNameEng',
```
Would it be possible to change these to just `familyName` so dropping the `Eng` and `En` suffixes?

Siyasanga replied:
Hi @rikukissa this is the [problem](https://github.com/opencrvs/opencrvs-core/blob/a124bb6a67ffa6745dde96429f96d7ef9858a357/packages/client/src/tests/util.tsx#L1445-L1461) that I was having with the test in relation to the request of renaming `familyNameEng` suffixes. In the mocked data example we already had `familyName` property and so trying to rename familyNameEng to `FamilyName` causes conflicts for the object. 

Since we are not using it in the `user-mgnt` database anyways I thought deleting it `familyName: 'হোসেন'` and renaming the other to   `familyName: 'Hossain'` would make sense would you agree?

rikukissa replied:
I agree 👍

Commit:
Remove lang suffix for FamilyName field

We no longer need the suffixes for the FamilyName field since we not really using it anywhere

https://github.com/opencrvs/opencrvs-core/issues/6830

Commit:
Add migration to remove NID field from user

A requirement from client to remove the NID field since most of the contries are not really using it.

https://github.com/opencrvs/opencrvs-core/issues/6830

Zangetsu101 commented on packages/client/src/forms/user/fieldDefinitions/createUser.ts:
```diff
@@ -79,38 +78,38 @@ export function userSectionFormType(): ISerializedFormSection {
             type: TEXT,
-            label: userFormMessages.firstNameEn,
+            label: userFormMessages.lastName,
             required: true,
             initialValue: '',
             validator: [{ operation: 'englishOnlyNameFormat' }],
             mapping: {
               mutation: {
                 operation: 'fieldToNameTransformer',
-                parameters: ['en', 'firstNames']
+                parameters: ['en', 'familyName']
               },
               query: {
                 operation: 'nameToFieldTransformer',
-                parameters: ['en', 'firstNames']
+                parameters: ['en', 'familyName']
               }
             }
           },
           {
-            name: 'familyNameEng',
+            name: 'firstNamesEng',
```
any particular reason why one's called "familyName" while the other "firstNamesEng"? :sweat_smile:

Siyasanga replied:
Thanks @Zangetsu101, good question. I may have misunderstood @rikukissa requested change from above. How I understood it was that a person can have multiple names from different languages, is this interpretation correct?

rikukissa replied:
@Siyasanga Sorry, I should have been clearer! A user can only have one name, always in English. The `-Eng` suffix is technical debt from the past :) If we need to have multiple names in different scripts later on we can build a new solution. So for now, all `En` and `Eng` suffixes should be removed both from core and country config client.csv

Zangetsu101 reviewed (commented):
Looks good otherwise :+1:

Commit:
Record changes in the CHANGELOG

To keep track of changes done on this PR

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Remove Eng suffix for the firstName in user form

We currently don't support milti language support for both surname and
first names so the ENG/EN suffixes are not needed. Also remove EN bits
that were left for the surname field

https://github.com/opencrvs/opencrvs-core/issues/6830
